src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/CryptoConfig.cs

```diff
@@ -301,17 +301,15 @@ public partial class CryptoConfig
 #endif
 
         [UnsupportedOSPlatform("browser")]
-        public static void AddAlgorithm(Type algorithm, params string[] names)
+        public static void AddAlgorithm(Type algorithm, params string[] names!!)
         {
 #if BROWSER
             throw new PlatformNotSupportedException(SR.SystemSecurityCryptography_PlatformNotSupported);
 #else
-            if (algorithm == null)
-                throw new ArgumentNullException(nameof(algorithm));
+            ArgumentNullException.ThrowIfNull(algorithm);
             if (!algorithm.IsVisible)
                 throw new ArgumentException(SR.Cryptography_AlgorithmTypesMustBeVisible, nameof(algorithm));
-            if (names == null)
-                throw new ArgumentNullException(nameof(names));
+            ArgumentNullException.ThrowIfNull(algorithm);
 
```

Comment:
This seems a bit off.  Why not `(Type algorithm!!, params string[] names!!)`?

Either way, seems like we don't need to ThrowIfNull for the algorithm parameter, the second one is double-guaranteed to not throw (barring some very scary CLR bugs)

Reply:
I interpreted this case (and others like it) as wanting to ensure PNSE is thrown first `#if BROWSER`.

Reply:
With that said, perhaps we should entertain taking another pass over these where we make exception-ordering changes and document them as subtle breaking changes, allowing us to use the `!!` syntax across the board.

Reply:
> With that said, perhaps we should entertain taking another pass over these where we make exception-ordering changes and document them as subtle breaking changes, allowing us to use the !! syntax across the board.

Yes, we should.  I tried in the vast majority of cases to avoid introducing any ordering changes as part of _this_  change, with the exception I called out in the PR description of allowing them in the relatively rare case where you had things like:
```C#
if (arg1 is null) throw new ArgumentNullException("arg1");
if (arg1.Count == 0) throw new ArgumentException(...);
if (arg2 is null) throw new ArgumentNullException("arg2");
if (arg2.Count == 0) throw new ArgumentException(...);
```
and being ok with changing that to:
```C#
if (arg1 is null) throw new ArgumentNullException("arg1");
if (arg2 is null) throw new ArgumentNullException("arg2");
if (arg1.Count == 0) throw new ArgumentException(...);
if (arg2.Count == 0) throw new ArgumentException(...);
```
(basically reordering some ArgumentExceptions when there were multiple validations of the same set of arguments).

That said, this particular example was a mistake (I still need to re-re-review the whole set of changes and fix some things I know are off).  The intent here was to undo the auto-fix and use ThrowIfNull manually rather than adding those checks in for browser, but I apparently neglected to remove the second `!!` the fixer had added.

Reply:
If we wanted to keep the PNSE first, then we'd presumably not use any `!!` here, and the other methods in this class would presumably do the same.

I'd had a similar "are we OK with the exception order changing?" thought at things like `new AesCcm((byte[])null)` when CCM isn't available.  Then I decided it doesn't really matter...  ANE is always supposed to be "your bug".  PNSE is "sorry, this isn't available due to circumstances you can't change".  I mean... sure... _maybe_ someone noticed they can do `new AesCcm(null)` and check PNSE vs ANE... but they should just have been checking AesCcm.IsSupported.  Similarly, here the only sensible reason to care about the ordering would be already answered by `OperatingSystem.IsBrowser()`.

As for this block, though, I'd expect either 0 or 2 uses of `!!`, not 1, and if we're doing a manual call to `ArgumentNullException.ThrowIfNull(param)` that we wouldn't have two for the same param :smile:.